tests/gitlab-single-node-asg/main.tf

```diff
@@ -0,0 +1,116 @@
+/** 
+ * Run a Gitlab on a single EC2 instance.
+ * This instance will be part of a single-node autoscaling group
+ * that shares an EBS volume to store data.
+ * 
+ * Note that there is a peculiarity with the EBS volume in that it
+ * requires some manual setup the very first time to make it available
+ * for use (unless a snapshot id is supplied):
+ *
+ * parted --script /dev/xvdf -- mklabel msdos
+ * parted --script /dev/xvdf -- mkpart primary 0 -1
+ * mkfs -t ext4 -F /dev/xvdf1
+ * e2label /dev/xvdf1 gitlab
+ *
+ * After running the above code to initialise the EBS, terminate the instance
+ * and the autoscaling group will bring up a new instance that will be running
+ * gitlab once it is done initialising.
+ */
+
+module "snasg" {
+  source                  = "../../tf-modules/single-node-asg"
+  name                    = "test"
+  az                      = "${var.az}"
+  key_name                = "${var.key_name}"
+  key_file                = "${var.key_file}"
+  ami                     = "${var.instance_ami}"
+  instance_type           = "${var.instance_type}"
+  name_suffix             = "gitlab-snasg"
+  subnet_id               = "${var.subnet_id}"
+  security_group_ids      = ["${aws_security_group.snasg-security-group.id}"]
+  region                  = "${var.region}"
```

Comment:
you can also use the `aws_region` `data` source to retrieve details like this, see https://www.terraform.io/docs/providers/aws/d/region.html

```
data "aws_region" "current" {
  current = true
}
```

Reply:
I'm not sure whether this is a good idea, given that you then 1. cannot have things in multiple regions at the same time and 2. include an assumption that the availability zone you supply is in this region.

Let me know if you still recommend this `data` approach, and I'll fix it.